lib/group_domain_models/lib/src/models/course.dart

```diff
@@ -307,12 +305,12 @@ class CourseData {
 class CourseSettings {
   final bool isPublic;
   final bool isMeetingEnabled;
-  final WritePermission writePermission;
+  final WritePermission? writePermission;
```

Comment:
Couldn't this be non-nullable and be the most restrictive option if none are given? Or is this on purpose because e.g. the backend assigns it and we don't want to set it?  

I'm just thinking that if some logic needs this value but its `null` then the only sensible option would be to interpret this as the most restrictive WritePermissions anyways, so we could make it non-nullable and implement it here.  
Idk if this makes sense, I'm not too familiar with this code.